esp/platform/persistent.cpp

```diff
@@ -86,9 +88,9 @@ class CAvailKeeper : implements IInterface, public CInterface
             auto iter = sset->begin();
             if (iter != sset->end())
```

Comment:
There're 2 problems with removing empty SocketSet: (1) it has to be re-allocated again if the same ep shows up after deletion (2) to delete it, will have to iterate through the map to find it, which can potentially be expensive. Considering the unlikeliness of having huge number of used-once ep's, I tend to think it's better not to clean up. What do you think @jakesmith ?

Reply:
2nd issue can be solved by getting hold of the ep from the sockinfo before erasing, so the major concern (#2 above) is not valid. I'll do the clean up. @jakesmith

Reply:
I did some performance test and found that removing empty socket sets seems to have a non-negligible impact on the throughput. I think that's because when the sockets are busy and get picked up right after they're put back, so the socket set gets empty all the time, so has to be constantly re-created. So for now I'm thinking not to do the removal. What do you think @jakesmith ?

Reply:
right. If there aren't expected to be an ever growing number, then prob. ok.
You could alternatively, look at the size on add, and when hits some high threshold walk and remove empty ones.
@mayx

Reply:
OK, let me do that just in case.

Reply:
What do you think would be a good threshold, 1000?